tests/ut/runtime/test_benchmark.py

```
@@ -208,6 +229,164 @@ def test_mean_tree_spread_annotations(require_simpler):
+    assert [d.task for d in r0_rank0] == ["a", "b"]
+    assert [d.device_wall_us for d in r0_rank0] == [4.0, 6.0]
+    # Round 1, rank 101: single dispatch (task c), device wall 8.
+    r1_rank1 = stats.rounds_dispatches[1][101]
+    assert [d.task for d in r1_rank1] == ["c"]
+    assert r1_rank1[0].device_wall_us == 8.0
+    # The nested view sums to the per-rank per-round busy figures.
+    assert stats.per_rank("device")[100] == [10.0, 10.0]  # 4+6, 3+7
+
+
+def test_parse_l3_per_rank_effective_time(require_simpler):
+    """Per-card L2 Effective = orch∪sched window; exposed per dispatch and per rank."""
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF002)